### PR TITLE
4410-feat(front): Implement Confirmation Prompt for Multiple Record Deletion

### LINKED ISSUE #4410: Implement Confirmation Prompt for Multiple Record Deletion
# Current Behavior

Users can delete multiple records without confirmation, risking unintentional removal or misunderstanding of the consequences. This process lacks a confirmation step, which could lead to irreversible data loss.

# Desired Behavior

A confirmation prompt should appear before deletion, ensuring the user knows several records, not just visible ones, are going to be deleted.

image

# Figma

https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=25201-46329&mode=design&t=Lc9vv0ssYD3TZhft-11

### PR DESCRIPTION
**Pull request by Anchit1909**
Closes #4410 

**Changes:**

- Added Confirmation Prompt for Multiple Record Deletion.

https://github.com/twentyhq/twenty/assets/22574091/d8836256-5156-49ea-b8af-a855d2531304

### CODE REVIEW

**Comment by Kanav-Arora** [comment]
Hi @Anchit1909 
Thanks for contributing! I'm leaving some comments suggesting some followup changes.

File: packages/twenty-front/src/modules/ui/navigation/action-bar/components/ActionBarItem.tsx [hunk 1]
```diff
@@ -80,13 +88,33 @@ export const ActionBarItem = ({ item }: ActionBarItemProps) => {
...
+                selectedRecords === 1 ? `record` : 'records'
+              }`}
+              subtitle={`This action cannot be undone. This will permanently delete ${
+                selectedRecords === 1 ? 'this record' : 'these records'
+              }`}
+              onConfirmClick={() => item.onClick?.()}
+              deleteButtonText="Delete Records"
+            />
+          )}
+        </>
```
Comment (Kanav-Arora) [inline packages/twenty-front/src/modules/ui/navigation/action-bar/components/ActionBarItem.tsx:117]: This doesn't seem to be a good approach as the base action for Delete is already implemented and conditional approach isn't good. We can keep the ActionBarItem code as it is. Instead please make changes to `useRecordActionBar`

**Review by Kanav-Arora — CHANGES_REQUESTED** [review]

**Review by Kanav-Arora — APPROVED** [review]

File: packages/twenty-front/src/modules/ui/navigation/action-bar/components/ActionBar.tsx [hunk 2]
```diff
@@ -17,14 +17,12 @@ const StyledContainerActionBar = styled.div`
   display: flex;
   height: 48px;
 
-  left: 50%;
+  left: 40%;
```
[the remark this answers is not in the document]
Comment (Anchit1909) [reply packages/twenty-front/src/modules/ui/navigation/action-bar/components/ActionBar.tsx:20]: `Left: 50%` and `transform: translateX(-50%);` centers an absolute positioned element. But after adding the confirmation modal, the size of the modal reduced to the size of the actionbar. To address this issue, I removed `transform: translateX(-50%);` and decreased the percentage of Left to 40% to center the element on the screen.

File: packages/twenty-front/src/modules/ui/navigation/action-bar/components/ActionBar.tsx [hunk 3]
```diff
@@ -17,14 +17,12 @@ const StyledContainerActionBar = styled.div`
...
   padding-left: ${({ theme }) => theme.spacing(2)};
   padding-right: ${({ theme }) => theme.spacing(2)};
   position: absolute;
   top: auto;
-
-  transform: translateX(-50%);
-  z-index: 1;
```
[the remark this answers is not in the document]
Comment (Anchit1909) [reply packages/twenty-front/src/modules/ui/navigation/action-bar/components/ActionBar.tsx:27]: I had increased the value of z-index to ensure the modal appears above all other elements.

File: packages/twenty-front/src/modules/ui/navigation/action-bar/components/ActionBarItem.tsx (hunk 1 above)
Comment (Anchit1909) [reply packages/twenty-front/src/modules/ui/navigation/action-bar/components/ActionBarItem.tsx]: tbh, i had doubts regarding my implementation. So, thanks for the review. I have made the necessary changes. Is this what you meant?

File: packages/twenty-front/src/modules/ui/navigation/action-bar/components/ActionBar.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Kanav-Arora) [reply packages/twenty-front/src/modules/ui/navigation/action-bar/components/ActionBar.tsx:20]: Ohk maybe we can proceed as there's no specific design regarding this

File: packages/twenty-front/src/modules/ui/navigation/action-bar/components/ActionBarItem.tsx (hunk 1 above)
Comment (Kanav-Arora) [reply packages/twenty-front/src/modules/ui/navigation/action-bar/components/ActionBarItem.tsx]: Yes basically I was suggesting reusability and not hard coded.

**Review by Kanav-Arora — CHANGES_REQUESTED** [review]
Hi @Anchit1909 
After discussions some further changes are required:
- For single record its showing 1 record(s) so we can fix the plurality
- In the action bar in case of single record selection please show `Delete` and not `Delete (1)`
- Moreover regarding the left:40% and transform-x, I'll look into it as in why is it required and what's the other way around.

For any doubts feel free to connect

@Bonapara could you please confirm this whether any design changes will be required for modal. I'll update you once I find the issue.

**Comment by Anchit1909** [comment]
> Hi @Anchit1909 After discussions some further changes are required:
> 
> * For single record its showing 1 record(s) so we can fix the plurality
> * In the action bar in case of single record selection please show `Delete` and not `Delete (1)`
> * Moreover regarding the left:40% and transform-x, I'll look into it as in why is it required and what's the other way around.
> 
> For any doubts feel free to connect
> 
> @Bonapara could you please confirm this whether any design changes will be required for modal. I'll update you once I find the issue.

Hey @Kanav-Arora,
Implemented the required changes. Lmk if you need anything else.

**Comment by Kanav-Arora** [comment]
Hi @Anchit1909 
Thanks for the changes. I'm dropping some comments

File: packages/twenty-front/src/modules/ui/navigation/action-bar/components/ActionBarItem.tsx [hunk 4]
```diff
@@ -80,13 +80,16 @@ export const ActionBarItem = ({ item }: ActionBarItemProps) => {
...
-        </StyledButton>
+        <>
+          <StyledButton
+            accent={item.accent ?? 'default'}
+            onClick={() => item.onClick?.()}
+          >
+            {item.Icon && <item.Icon size={theme.icon.size.md} />}
+            <StyledButtonLabel>{item.label}</StyledButtonLabel>
+          </StyledButton>
+          {item.ConfirmationModal}
```
Comment (Kanav-Arora) [inline packages/twenty-front/src/modules/ui/navigation/action-bar/components/ActionBarItem.tsx:91]: Now this needs to be fixed you can either wrap it in a div but its giving error as the modal is relative to actionbaritem component. I hope you are getting the issue. Can you please fix this? If you need any help please ping

**Review by Kanav-Arora — CHANGES_REQUESTED** [review]

File: packages/twenty-front/src/modules/ui/navigation/action-bar/components/ActionBarItem.tsx (hunk 4 above)
Comment (Kanav-Arora) [reply packages/twenty-front/src/modules/ui/navigation/action-bar/components/ActionBarItem.tsx:91]: Hi @Anchit1909 
For the numbering, I think its weird that the width is changing due to change in number of selected rows.

@Bonapara has made some changes in the design 

https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=13736-50538&mode=design&t=xKOKPr1F1nQcJSIR-11

Comment (Bonapara) [reply packages/twenty-front/src/modules/ui/navigation/action-bar/components/ActionBarItem.tsx:91]: Hi @Kanav-Arora! I tried Figma, and I believe keeping the ":" makes it clearer that the actions on the left are the ones targeted by the selection :)

Comment (Kanav-Arora) [reply packages/twenty-front/src/modules/ui/navigation/action-bar/components/ActionBarItem.tsx:91]: Thanks Thomas

@Anchit1909 please proceed with the implementation of figma design

Comment (Anchit1909) [reply packages/twenty-front/src/modules/ui/navigation/action-bar/components/ActionBarItem.tsx]: Hey @Kanav-Arora , 
I have implemented the new design. 
Also, I tried multiple ways to center the ActionBar without separating the ConfirmationModal, but nothing worked. If I don't separate the ConfirmationModal, it attaches itself to the actionbar and does not take the whole screen as it is absolutely positioned.

Please let me know if this approach is correct or if there is something else that I should do.

**Review by Kanav-Arora — APPROVED** [review]

**Comment by FelixMalfait** [comment]
Thanks @Kanav-Arora! I think ConfirmationModal is correct since it's a component that's the convention used across the app

**Review by FelixMalfait — APPROVED** [review]
Great work! Thanks!

**Comment by Kanav-Arora** [comment]
Great work @Anchit1909 ! Thanks for contributing

**Pull request merged**